public/scripts/group-chats.js

```diff
@@ -857,14 +859,21 @@ async function generateGroupWrapper(by_auto_mode, type = null, params = {}) {
             await saveChatConditional();
             $('#send_textarea').val('')[0].dispatchEvent(new Event('input', { bubbles:true }));
         }
-
+        if (showChatQueue){
+            for (let i = 0; i < activatedMembers.length; ++i){
+                characters[activatedMembers[i]].queueOrder = (i+1);
```

Comment:
I would suggest storing the queue order in some other place than the character object.
The character is a global representation of the char card, and is not related to any chat really.  
This might give the wrong impression that this data is directly related to the character, and not the group member in the context of this group chat.

You can either think about just using a local Map inside the `group-chats.js` file, or maybe even saving the queue order into the chat context, if you want it persisted through possible reloads of the page. Though I think the latter doesn't make sense, because with a refresh generation stops anyway, right?

Reply:
This should not be leaving the generateGroupWrapper function even. By keeping a local map of the draft and passing it to printGroupMembers (default value = empty map) it is possible to keep the JSON data clean and not create any more global variables.

Reply:
Removed it from the character object.  I wasn't able to keep it local to generateGroupWrapper though.  There's an issue there when you pop out the chat members while a response is being generated.  The pop out runs through printGroupMembers, but knew nothing about the queue order determined during the draft so highlighting did not function.